zipline-gradle-plugin/src/main/kotlin/app/cash/zipline/gradle/ZiplinePlugin.kt

```diff
@@ -77,11 +79,15 @@ class ZiplinePlugin : KotlinCompilerPluginSupportPlugin {
       }
     }
 
+    val ziplineApiCheck = target.tasks.register("ziplineApiCheck")
+    target.tasks.named("check").configure { checkTask ->
+      checkTask.dependsOn(ziplineApiCheck)
+    }
+    val ziplineApiDump = target.tasks.register("ziplineApiDump")
     target.tasks.withType(KotlinCompile::class.java) { kotlinCompile ->
-      if (kotlinCompile.name == "compileKotlinJvm") {
-        registerZiplineApiTask(target, kotlinCompile, ZiplineApiValidationTask.Mode.Check)
-        registerZiplineApiTask(target, kotlinCompile, ZiplineApiValidationTask.Mode.Dump)
-      }
+      if ("Test" in kotlinCompile.name) return@withType
```

Comment:
I think we can get at the main compilation task through
```kotlin
kotlinExtension.targets.all { target -> .. }
```
but that can be a follow-up. At the very least it reduces the risk of name-based task logic. Someone could, in theory, have "Test" in their target name.

Reply:
Yeah, lemme do